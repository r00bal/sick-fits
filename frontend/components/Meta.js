import Head from 'next/head';

const Meta = () => (
  <Head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />

    <link rel="shortcut icon" href="/static/favicon.png" type="image/x-icon" />
    <link rel="stylesheet" href="/static/nprogress.css" />
    <title>Sick Fits!</title>
  </Head>
);

export default Meta;
