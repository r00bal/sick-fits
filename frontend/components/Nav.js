import Link from 'next/link';
import NavStyles from './styles/NavStyles';
import Sell from '../pages/sell';
import Router from 'next/router';

const Nav = () => (
  <NavStyles>
    <Link href="/items">
      <a>Items</a>
    </Link>
    <Link href="/sell">
      <a>Sell</a>
    </Link>
    <Link href="/signup">
      <a>Signup</a>
    </Link>
    <Link href="/orders">
      <a>Orders</a>
    </Link>
    <Link href="/me">
      <a>Me</a>
    </Link>
  </NavStyles>
);

export default Nav;
