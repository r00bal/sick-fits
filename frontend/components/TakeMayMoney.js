import React, { Component } from 'react';
import StripCheckout from 'react-stripe-checkout';
import { Mutation } from 'react-apollo';
import Router from 'next/router';
import NProgress from 'nprogress';
import PropTypes from 'prop-types';
import gql from 'graphql-tag';
import calcTotalPrice from '../lib/calcTotalPrice';
import Error from './ErrorMessage';
import User, { CURRENT_USER_QUERY } from './User';

function totalItems(cart) {
  return cart.reduce((tally, cartItem) => tally + cartItem.quantity, 0);
}

class TakeMayMoney extends Component {
  onToken = res => {
    console.log('On Token Called!');
    console.log(res);
  };

  render() {
    return (
      <User>
        {({ data: { me } }) => (
          <StripCheckout
            amount={calcTotalPrice(me.cart)}
            name="Sick Fits"
            description={`Order of ${totalItems(me.cart)} items`}
            image={me.cart[0].item && me.cart[0].item.image}
            stripeKey="pk_test_KBbKNVsKRnLlyj1fvRoFZ8FQ00mZnxADUJ"
            currency="USD"
            email={me.email}
            token={res => this.onToken(res)}>
            {this.props.children}
          </StripCheckout>
        )}
      </User>
    );
  }
}

export default TakeMayMoney;
