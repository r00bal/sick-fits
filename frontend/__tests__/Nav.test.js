import Nav from '../components/Nav';
import { CURRENT_USER_QUERY } from '../components/User';
import { shallow, mount } from 'enzyme';
import wait from 'waait';
import toJSON from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';
import { fakeUser, fakeCartItem } from '../lib/testUtils'

const notSignedInMocks = [
 {
  request: { query: CURRENT_USER_QUERY },
  result: { data: { me: null } }
 },
];

const signedInMocks = [
 {
  request: { query: CURRENT_USER_QUERY },
  result: { data: { me: fakeUser() } }
 },
];

const signedInMocksWithCartItems = [
 {
  request: { query: CURRENT_USER_QUERY },
  result: {
   data: {
    me: {
     ...fakeUser(),
     cart: [fakeCartItem(), fakeCartItem(), fakeCartItem(),]
    }
   }
  }
 },
];

describe('<Nav />', () => {
 it('renders the minimal nav when signed out', async () => {
  const wrapper = mount(
   <MockedProvider mocks={notSignedInMocks}>
    <Nav />
   </MockedProvider>
  );
  await wait();
  wrapper.update();
  const nav = wrapper.find('[data-test="nav"]');
  expect(toJSON(nav)).toMatchSnapshot();
 })

 it('it renders full nav when signed in', async () => {
  const wrapper = mount(
   <MockedProvider mocks={signedInMocks}>
    <Nav />
   </MockedProvider>
  );
  await wait();
  wrapper.update();
  const nav = wrapper.find('ul[data-test="nav"]');
  expect(nav.children().length).toBe(6);
  // expect(toJSON(nav)).toMatchSnapshot();
 })

 it('it renders the amount of items in the cart', async () => {
  const wrapper = mount(
   <MockedProvider mocks={signedInMocksWithCartItems}>
    <Nav />
   </MockedProvider>
  );
  await wait();
  wrapper.update();
  const nav = wrapper.find('[data-test="nav"]');
  const count = nav.find('div.count');
  expect(toJSON(count)).toMatchSnapshot();
  // expect(toJSON(nav)).toMatchSnapshot();
 })
})
