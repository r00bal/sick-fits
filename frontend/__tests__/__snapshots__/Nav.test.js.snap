// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> it renders the amount of items in the cart 1`] = `
<div
  className="count CartCount__Dot-xxvp4g-1 kudjIL"
>
  9
</div>
`;

exports[`<Nav /> renders the minimal nav when signed out 1`] = `
Array [
  <NavStyles
    data-test="nav"
  >
    <ul
      className="NavStyles-sc-11c0d2g-0 kNymmY"
      data-test="nav"
    >
      <Link
        href="/items"
      >
        <a
          href="/items"
          onClick={[Function]}
        >
          Shop
        </a>
      </Link>
      <Link
        href="/signup"
      >
        <a
          href="/signup"
          onClick={[Function]}
        >
          Signup
        </a>
      </Link>
    </ul>
  </NavStyles>,
  <ul
    className="NavStyles-sc-11c0d2g-0 kNymmY"
    data-test="nav"
  >
    <Link
      href="/items"
    >
      <a
        href="/items"
        onClick={[Function]}
      >
        Shop
      </a>
    </Link>
    <Link
      href="/signup"
    >
      <a
        href="/signup"
        onClick={[Function]}
      >
        Signup
      </a>
    </Link>
  </ul>,
]
`;
